### DISCUSSION TITLE
Is there any kinds of attached pseudo class support in plan?

### DISCUSSION THREAD

**Discussion by Flithor** [AvaloniaUI/Avalonia#11890] [Ideas]
It seems that the pseudo-class can only be extended by inheritance.

**Comment by maxkatz6** [comment]
No, pseudo-classes set from the outside of the control are not planned.
You can use style classes instead.

**Reply by Flithor** [reply]
So If I want to add a `CheckBox` in `ListBoxItem` template and apply `ListBoxItem:checked` style, I need inherit ListBoxItem and add new pseudo class?
Trigger in WPF would be very easy to implement the same result, even without inheritance.

As I know, in Avalonia 0.10, there is no way to inherit the control style and override some property. Inherit an existed control looks painful.

**Reply by rabbitism** [reply]
You can just set IsChecked to the value you need.

**Reply by Flithor** [reply]
> You can just set IsChecked to the value you need.

@rabbitism Hey, what? ListBoxItem has no property named `IsChecked`

**Reply by rabbitism** [reply]
> @rabbitism Hey, what? ListBoxItem has no property named `IsChecked`

It has IsSelected property.

**Reply by Flithor** [reply]
> > @rabbitism Hey, what? ListBoxItem has no property named `IsChecked`
> 
> It has IsSelected property.

Well, but I'm not want to let "Check" do same work as "Select".

**Reply by rabbitism** [reply]
Can you explain what you want to achieve?

**Reply by Flithor** [reply]
Oh~ I can move the `CheckBox` into `DataTemplate`!
I'm just digging into a dead end.
